src/LoadingBar.js

```diff
@@ -0,0 +1,124 @@
+const React = require('react');
+const { bool } = React.PropTypes;
+
+/**
+ * Displays a loading bar (YouTube-like) at the top of container
+ * Based on https://github.com/lonelyclick/react-loading-bar/blob/master/src/Loading.jsx
+ *
+ * @type {ReactClass}
+ */
+const LoadingBar = React.createClass({
+    propTypes: {
+        show:  bool
+    },
+
+    getDefaultProps() {
+        return {
+            show: false
+        };
+    },
+
+    getInitialState() {
+        return {
+            size: 0,
+            disappearDelayHide: false, // when dispappear, first transition then display none
+            percent: 0,
+            appearDelayWidth: 0 // when appear, first display block then transition width
+        };
+    },
+
+    componentWillReceiveProps(nextProps) {
+        const { show } = nextProps;
+
+        if (show) {
+            this.show();
+        } else {
+            this.hide();
+        }
+    },
+
+    shouldComponentUpdate(nextProps, nextState) {
+        return true; // !shallowEqual(nextState, this.state)
+    },
+
+    show() {
+        let { size, percent } = this.state;
+
+        const appearDelayWidth = size === 0;
+        percent = calculatePercent(percent);
+
+        this.setState({
+            size: ++size,
+            appearDelayWidth,
+            percent
+        });
+
+        if (appearDelayWidth) {
+            setTimeout(() => {
+                this.setState({
+                    appearDelayWidth: false
+                });
+            });
+        }
+    },
+
+    hide() {
+        let { size } = this.state;
+
+        if (--size < 0) {
```

Comment:
Dangerous modification of the `size` variable.
Can you do a `if ((size - 1) < 0)` instead?